crates/forge_main/src/ui.rs

```diff
@@ -628,7 +643,12 @@ impl<A: API + 'static, F: Fn() -> A> UI<A, F> {
     async fn init_state(&mut self, first: bool) -> Result<Workflow> {
         let provider = self.init_provider().await?;
         let mut workflow = self.api.read_workflow(self.cli.workflow.as_deref()).await?;
-        if workflow.model.is_none() {
+        let model_available = match workflow.model.as_ref() {
+            Some(m) => self.is_model_available(m).await.unwrap_or(false),
+            None => false,
+        };
```

Comment:
Instead of proactively checking, we can lazily check once the error response is received.

Reply:
We are checking from the cached response. I think, it's better ux if people change the provider and model in forge.yaml doesn't exist they will be able to choose one.

Thoughts?